File: user-role-service/models/userModel.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/db');

// Defines a User model for the 'users' table
const User = sequelize.define('users', {
    id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true // Auto-incrementing primary key
    },
    username: {
        type: DataTypes.STRING,
        allowNull: false // Username is required
    },
    email: {
        type: DataTypes.STRING,
        allowNull: false // Email is required
    },
    password_hash: {
        type: DataTypes.STRING,
        allowNull: false // Password hash is required
    }
}, {
    tableName: 'users', // Specifies the table name
    timestamps: false // Disables automatic timestamp columns
});

module.exports = User;

File: read-user/models/user_model.py
from config.db import db

class User(db.Model):
    """
    Represents a User in the database.

    Attributes:
        id (int): The primary key of the user.
        username (str): The username of the user.
        email (str): The email address of the user.
        password_hash (str): The hashed password of the user.
    """

    __tablename__ = 'users'

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(50), nullable=False)
    email = db.Column(db.String(100), nullable=False)
    password_hash = db.Column(db.String(255), nullable=False)

    def to_dict(self):
        """
        Convert the User instance to a dictionary.

        Returns:
            dict: A dictionary representation of the User instance.
        """
        return {
            'id': self.id,
            'username': self.username,
            'email': self.email
        }
